packages/gatsby/src/commands/develop.ts

```diff
@@ -1,631 +1,240 @@
-
-  // report.stateUpdate(`webpack`, `IN_PROGRESS`)
-
-  const webpackActivity = report.activityTimer(`Building development bundle`, {
-    id: `webpack-develop`,
-  })
-  webpackActivity.start()
-
-  const devConfig = await webpackConfig(
-    program,
-    directory,
-    `develop`,
-    program.port,
-    { parentSpan: webpackActivity.span }
+  const oldConfigString = JSON.stringify(
+    { ...lastConfig, siteMetadata: null },
+    replacer
   )
-
-  const compiler = webpack(devConfig)
-
-  /**
-   * Set up the express app.
-   **/
-  const app = express()
-  app.use(telemetry.expressMiddleware(`DEVELOP`))
-  app.use(
-    webpackHotMiddleware(compiler, {
-      log: false,
-      path: `/__webpack_hmr`,
-      heartbeat: 10 * 1000,
-    })
+  const newConfigString = JSON.stringify(
+    { ...newConfig, siteMetadata: null },
+    replacer
   )
 
-  app.use(cors())
+  if (oldConfigString === newConfigString) return false
 
-  /**
-   * Pattern matching all endpoints with graphql or graphiql with 1 or more leading underscores
-   */
-  const graphqlEndpoint = `/_+graphi?ql`
+  return true
+}
 
-  if (process.env.GATSBY_GRAPHQL_IDE === `playground`) {
-    app.get(
-      graphqlEndpoint,
-      graphqlPlayground({
-        endpoint: `/___graphql`,
-      }),
-      () => {}
-    )
-  } else {
-    graphiqlExplorer(app, {
-      graphqlEndpoint,
-    })
+class ControllableScript {
+  private process
+  private script
+  public isRunning
+  constructor(script) {
+    this.script = script
   }
-
-  app.use(
-    graphqlEndpoint,
-    graphqlHTTP(
-      (): graphqlHTTP.OptionsData => {
-        const {
-          schema,
-          schemaCustomization,
-        }: {
-          schema: GraphQLSchema
-          schemaCustomization: {
-            composer: SchemaComposer<any>
-            context: any
-          }
-        } = store.getState()
-
-        return {
-          schema,
-          graphiql: false,
-          context: withResolverContext({
-            schema,
-            schemaComposer: schemaCustomization.composer,
-            context: {},
-            customContext: schemaCustomization.context,
-          }),
-          customFormatErrorFn(err): unknown {
-            return {
-              ...formatError(err),
-              stack: err.stack ? err.stack.split(`\n`) : [],
-            }
-          },
-        }
-      }
-    )
-  )
-
-  /**
-   * Refresh external data sources.
-   * This behavior is disabled by default, but the ENABLE_GATSBY_REFRESH_ENDPOINT env var enables it
-   * If no GATSBY_REFRESH_TOKEN env var is available, then no Authorization header is required
-   **/
-  const REFRESH_ENDPOINT = `/__refresh`
-  const refresh = async (req: express.Request): Promise<void> => {
-    stopSchemaHotReloader()
-    let activity = report.activityTimer(`createSchemaCustomization`, {})
-    activity.start()
-    await createSchemaCustomization({
-      refresh: true,
+  start(): void {
+    const tmpFileName = tmp.tmpNameSync({
```

Comment:
I rather not use tmp for something we can easily do ourselves, or maybe I'm missing something here.

Reply:
There is a lot of logic in this (e.g. retries to find a unique name in case that specific one is taken already) that we don't really want to replicate! The package has 17M weekly downloads, so even though the version might be 0. I think we are fine to rely on it.

Reply:
Jup I saw, I thought it was just a wrapper around os.tmpdir() & fs.mkTemp or something.

We'll have to add the package to renovate to make sure we don't upgrade it https://github.com/gatsbyjs/gatsby/blob/master/renovate.json5#L83-L114